akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -247,10 +246,15 @@ private[akka] class Association(
   }
 
   private def runOutboundOrdinaryMessagesStream(): Unit = {
-    val (q, completed) = Source.queue(256, OverflowStrategy.dropBuffer)
+    // FIXME config queue size, and it should also be possible to use some kind of LinkedQueue
+    //       for less memory consumption
+    val q = new ManyToOneConcurrentArrayQueue[Send](3072)
+    val (queueValue, completed) = Source.fromGraph(new SendQueue[Send])
       .toMat(transport.outbound(this))(Keep.both)
       .run()(materializer)
-    queue = q
+    queueValue.inject(q)
```

Comment:
@drewhk I have implemented the injection of the queue instead of returning it. I have not yet implemented the full lockless thing in the Association, but please confirm that this is in line with what you were thinking.